app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,31 @@
+class NewsletterFinder
+  include Enumerable
+
+  delegate :each, to: :newsletters_list
+
+  def initialize(type)
+    @type = type
+  end
+
+  private
+
+  def newsletters_list
+    @newsletters_list ||= newsletters_with_file_extension.map do |newsletter|
+      newsletter.gsub(/\.pdf/, "")
+    end
+  end
+
+  def newsletters_with_file_extension
+    unsorted_newsletters.sort do |date_1, date_2|
+      Date.parse("1-#{date_2}") <=> Date.parse("1-#{date_1}")
```

Comment:
Ah right, I don't frequently parse dates in that format so for some reason I thought the 15/16 in your example were days and 1 was a year, but this makes sense. I wouldn't worry about changing anything.